src/ATen/function_wrapper.py

```diff
@@ -480,7 +535,9 @@ def emit_body(env, option):
 
                     check_cast = CHECKED_CAST[arg['type']].substitute(
                         env, arg_name=arg['name'], arg_pos=count,
-                        null_okay=null_okay)
+                        null_okay=null_okay,
+                        # don't allow expansion to a scalar
+                        zero_dim_to_one='true' if not option['name'] == 'expand' else 'false')
```

Comment:
Can you explain what this is doing?

Reply:
This is to prevent expanding a size (1,) tensor to a size () tensor.

Let's look at the generated code before this change:
```
auto self_ = checked_cast<CPUFloatTensor>(self.pImpl,"self",1, false);
auto size_ = THLongStorageView::make(size, true);
return Tensor((new CPUFloatTensor(context,THFloatTensor_newExpand(self_->tensor, size_)))->maybeScalar(size.size() == 0),false);
```
the `true` in the call to `THLongStorageView` means convert a 0-d size to a 1-d size; so this would all pass and yield a 0-d tensor.  Now you get:
```
invalid argument 1: the number of sizes provided must be greater or equal to the number of dimensions in the tensor at /home/gchanan/local/ATen/lib/TH/generic/THTensor.c:298
```

Reply:
The problem with this, though, is making it `false` breaks the 0-dim expanding to 0-dim case (because in TH/THC the tensor is viewed as 1-d).  So, the choices seem to be:
1) add a one-off check that throws an exception for the 1-d expanding to 0-d case.
2) leave things as is, and eventually solve in TH/THC.